DatadogInternal/Sources/Telemetry/Telemetry.swift

```diff
@@ -98,15 +127,38 @@ public extension Telemetry {
         }
     }
 
-    /// Stops a method call, transforms method call metric to telemetry message,
-    /// and transmits on the message-bus of the core.
+    /// Stops timing a method call and posts a value for the "Method Called" metric.
+    ///
+    /// This method applies tail-based sampling in addition to the head-based sampling applied in `startMethodCalled()`.
+    /// The tail sample rate is compounded with the head sample rate and the telemetry sample rate to determine the effective sample rate.
     ///
-    /// - Parameters
+    /// - Parameters:
     ///   - metric: The `MethodCalledTrace` instance.
-    ///   - isSuccessful: A flag indicating if the method call was successful.
-    func stopMethodCalled(_ metric: MethodCalledTrace?, isSuccessful: Bool = true) {
+    ///   - isSuccessful: A flag indicating whether the method call was successful.
+    ///   - sampleRate: The sample rate for **tail-based** sampling of the metric, applied after the metric is processed by the SDK core.
+    ///     Defaults to `MetricTelemetry.defaultSampleRate` (15%).
+    func stopMethodCalled(
+        _ metric: MethodCalledTrace?,
+        isSuccessful: Bool = true,
+        sampleRate: Float = MetricTelemetry.defaultSampleRate
```

Comment:
Would it be clearer for customers to have this parameter named `tailBasedSampleRate` here, and `headBasedSampleRate` in `startMethodCalled`?

Reply:
Oh, this was great idea 👍 - I also found `sampleRate` confusing in both `start()` and `stop()` but changing to `headSampleRate` and `tailSampleRate` makes total sense here. Done ✅.

BTW, this is `DatadogInternal` API, so not meant for customer use.